### DISCUSSION TITLE
Automated generation of Python related best-of-lists

### DISCUSSION THREAD

**Discussion by fkromer** [best-of-lists/best-of#161] [Ideas]
It could be kind of cool to update best-of lists about Python packages in an automated manner. E.g. in case it's about a best-of list about `pydantic`:

- fetching [Python packages tagged with "pydantic" from pypi](https://pypi.org/search/?q=pydantic&o=&c=Framework+%3A%3A+Pydantic)
- fetching [Python packages tagged with "pydantic" from github](https://github.com/topics/pydantic)
- use the common set of packages to generate project sections (`project:`) in `projects.yml`
- use meta data of Python packages to generate category sections (`category:`) in `project.yml`
- get all Python packages from github, parse and check if the package (`pydantic`) is contained in dependencies (`requirements.txt`, `pyproject.toml`, ...)

I'm pretty sure it's not possible to fully automate the overall tasks of selecting and pasting packages into `pyproject.toml`. But getting potentially interesting packages using an approach like mentioned before could be pretty helpful probably.

What's your opinion about this idea @LukasMasuch ?

**Comment by lukasmasuch** [comment]
That would definitely be something very useful, and I probably also possible to do. I actually have some notebooks that contain a semi-automated way to collect packages, but it would still require a lot more time to get it full automated. Unforuntalty, I don't have a lot of time at the moment to work on new features for best-of :( If you want to work on this, maybe it would be nice to create that as a GitHub action as well.  E.g. something that automatically collects new packages, updates the project.yml and creates a new PR with the updated projects.

**Reply by fkromer** [reply]
I think I've some notebooks/scripts somewhere on my machine as well. We could collect those and make it public to the community. Whoever has time could pick up taking this effort further then.
